tests/operator_test.go

```diff
@@ -1150,6 +1150,55 @@ spec:
 		})
 	})
 
+	Describe("[test_id:unknown_id]should apply component configuration", func() {
+
+		It("test VirtualMachineInstancesPerNode", func() {
+			newVirtualMachineInstancesPerNode := 10
+
+			By("Updating KubeVirt Object")
+			kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			kv.Spec.Configuration.VirtualMachineInstancesPerNode = &newVirtualMachineInstancesPerNode
```

Comment:
To guarantee we're actually modifying the value, how about asserting it's nil before setting it?
```go
Expect(kv.Spec.Configuration.VirtualMachineInstancesPerNode).To(BeNil())
kv.Spec.Configuration.VirtualMachineInstancesPerNode = &newVirtualMachineInstancesPerNode
```

Reply:
Both options could fail (although your option is less likely to). Either way, I think both are fine as this failure will simply raise a flag that a test update is needed - which is better than silently/vacuously passing.

Reply:
I've just realized this comparison is incorrect, since it compares `*int` and `int` (which are deemed unequal by `reflect.DeepEqual()`. Use this:
```go
Expect(kv.Spec.Configuration.VirtualMachineInstancesPerNode).ToNot(Equal(&newVirtualMachineInstancesPerNode))
```